src/engine/strat_engine/engine.rs

```diff
@@ -283,6 +281,19 @@ impl Engine for StratEngine {
         }
     }
 
+    fn teardown_pool(&mut self, uuid: PoolUuid) -> StratisResult<bool> {
+        let (pool_name, pool) = match self.pools.remove_by_uuid(uuid) {
+            Some((name, pool)) => (name, pool),
+            None => return Ok(false),
+        };
+
+        let mut temp_table = Table::default();
+        temp_table.insert(pool_name.clone(), uuid, pool);
+        teardown_pools(temp_table)?;
```

Comment:
I'm expecting that you'll do something more scrupulous than this. There is also the problem that we have been dealing with recently of failing during some removal action, but removing it from the internal stratis data structures, so that there is a bad situation. So that would have to be dealt with somehow.